zbsprimarystorage/zbsprimarystorage/zbsutils.py

```
@@ -1,133 +1,243 @@
+        return shell.call(
+            "%s children --path %s/%s --user %s --format json" % (ZBS_BIN_PATH, logical_pool, volume, ZBS_USER_NAME))
+
+
+def is_support_get_volume_clients():
+    return shell.run("%s list client --help | grep -E '\--path'" % ZBS_BIN_PATH) == 0
+
+def is_volume_exist(logical_pool, volume):
+    o = query_volume_info(logical_pool, volume)
+    ret = jsonobject.loads(o)
+    if ret.error.code != 0:
+        match = re.search(r"status code:\s*(\w+)", ret.error.message)
```

Ruff (0.14.14)

[error] `re` may be undefined, or defined from star imports

(F405)